src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -187,7 +187,8 @@ internal static NuGetFramework GetTargetFramework(IVsProjectProperties propertie
                     targetFrameworkInfo.ProjectReferences
                         .Cast<IVsReferenceItem>()
                         .Where(IsReferenceOutputAssemblyTrueOrEmpty)
-                        .Select(item => ToProjectRestoreReference(item, projectDirectory)));
+                        .Select(item => ToProjectRestoreReference(item, projectDirectory))
+                        .Distinct(ProjectRestoreReferenceComparer.Default));
```

Comment:
Given that `ProjectRestoreReference` implements `IEquatable`, As per second example in https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct?view=net-6.0 it should work without passing custom comparer. It is possible that I may be missing an important scenario.

https://github.com/NuGet/NuGet.Client/blob/71424bcf70087a60abc6ec6b86035bd572400464/src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreReference.cs#L13

Reply:
ProjectRestoreReference's implementation compares not only project path/unique name, but also IncludeAssets, ExcludeAssets and PrivateAssets.  If for some reason these are different, then we'll have the same problem with there being duplicate references to the same project.